vault/request_handling.go

```diff
@@ -265,6 +265,23 @@ func (c *Core) checkToken(ctx context.Context, req *logical.Request, unauth bool
 		return nil, nil, errors.New("cannot access root path in unauthenticated request")
 	}
 
+	// At this point we won't be forwarding a raw request; we should delete
+	// authorization headers as appropriate
+	switch req.ClientTokenSource {
+	case logical.ClientTokenFromVaultHeader:
+		delete(req.Headers, consts.AuthHeaderName)
+	case logical.ClientTokenFromAuthzHeader:
+		if headers, ok := req.Headers["Authorization"]; ok {
+			for i, v := range headers {
+				if !strings.HasPrefix(v, "Bearer ") {
+					continue
+				}
+				headers = append(headers[:i], headers[i+1:]...)
```

Comment:
I believe this will append non Bearer headers multiple times, see: https://play.golang.org/p/HJatGEPsjF7

Reply:
Good point. Just pushed a change, can you 👀 ?

Reply:
Nice, looks better!